Firestore/core/src/remote/firebase_metadata_provider_apple.mm

```diff
@@ -52,8 +52,13 @@
 
 void FirebaseMetadataProviderApple::UpdateMetadata(
     grpc::ClientContext& context) {
+  int heartbeat = GetHeartbeat();
+  if (heartbeat == 0) {
+    return;
```

Comment:
optional: the method name looks generic enough for other thing unrelated to platform logs to be added there. This early exist may be very easy to miss, so the additional metadata may be unintentionally dropped. I would either wrap platform logs metadata into `if(heartbeat != FIRHeartbeatInfoCodeNone)` or introduce a separate method with a name like `addPlatformLoggingMetadata()` with the early exist which will be called from `UpdateMetadata()`. WDYT?

Reply:
This class is intended specifically for the Firebase platform logging-related functionality. We may abstract away other things (e.g. Cloud headers) in a similar class, but it would be a separate one.

Reply:
Thank you for clarification. IMO, the class and method names look a bit too generic in this case, but I don't feel too strong about it.